src/plugins/GreaseMonkey/gm_manager.cpp

```diff
@@ -256,7 +257,25 @@ void GM_Manager::scriptChanged()
 
 void GM_Manager::doDownloadScript(const QUrl &url)
 {
-    new GM_Downloader(url, this);
+    GM_Downloader* downloader = new GM_Downloader(url, this);
+    connect(downloader, &GM_Downloader::finished, this, [=](const QString& filename){
```

Comment:
-> `const QString &filename`
Also please change it to `fileName` as you are already using it in other places.